lib/src/rewrite.rs

```diff
@@ -600,8 +600,6 @@ impl<'settings, 'repo> DescendantRebaser<'settings, 'repo> {
         self.heads_to_remove.clear();
         self.heads_to_add.clear();
         self.mut_repo.set_view(view);
-        self.mut_repo.clear_rewritten_commits();
-        self.mut_repo.clear_abandoned_commits();
```

Comment:
I slightly prefer the original form because all the finalization code is at one place, but I'm not against changing that as things will be cleaned up later?

(I don't follow the discussion on the DescendantRebaser redesign, so ignore me if it doesn't make sense.)

Reply:
Firstly, this is very likely to be changed later (and is already changed slightly in the next commit), so I don't care very much what we do here now. If you insist, I can keep it.

I'm also not entirely sure what the end state will be right now. The major change I'm planning is to keep track of what's currently called `parent_mapping` from previous rebases, and to take it into account during future rebases.

I did have a reason to move it: currently, the rewritten and abandoned commits are passed to `DescendantRebaser::new`, and something else could be passed in their stead. It seems illogical to unconditionally clear these map in the latter case.

A possible alternative would be to remove these arguments to `DescendantRebaser::new` and have it always use these same maps as input, which would be fine, but would very likely need changes later.

Reply:
> currently, the rewritten and abandoned commits are passed to `DescendantRebaser::new`, and something else could be passed in their stead. It seems illogical to unconditionally clear these map in the latter case.

Yeah, it's a bit weird, but it's kinda okay now since the rebaser is constructed only by mut_repo? My little concern about moving things to the caller is that it's easy to forget to clear these mappings. 

Anyway, I don't have strong opinion about that.

Reply:
>  it's kinda okay now since the rebaser is constructed only by mut_repo?

Yes, especially after both of the PRs, where we are guaranteed that nobody else tries to construct the rebaser.

Again, I expect this to change later. I'm on the fence between doing nothing and implementing the "possible alternative" I described in the last comment; let me know if you have a better idea, or we can discuss this again later.

Reply:
btw, we still have tests who construct `DescendantRebaser` by `create_descendant_rebaser()`. Shouldn't they also need to do `clear_descendant_rebaser_plans()` if the rebaser returned `None`?

Reply:
This should not be an issue after #2738.

Reply:
> However, this should not be an issue after https://github.com/martinvonz/jj/pull/2738.

What do you think about moving this last commit to that PR then?

Reply:
Good idea, will do.